homeassistant/components/tado/device_tracker.py

```diff
@@ -40,15 +41,28 @@ async def async_get_scanner(
 ) -> DeviceScanner | None:
     """Configure the Tado device scanner."""
     device_config = config["device_tracker"]
-    import_result = await hass.config_entries.flow.async_init(
-        DOMAIN,
-        context={"source": SOURCE_IMPORT},
-        data={
-            CONF_USERNAME: device_config[CONF_USERNAME],
-            CONF_PASSWORD: device_config[CONF_PASSWORD],
-            CONF_HOME_ID: device_config.get(CONF_HOME_ID),
-        },
-    )
+    try:
+        import_result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_USERNAME: device_config[CONF_USERNAME],
+                CONF_PASSWORD: device_config[CONF_PASSWORD],
+                CONF_HOME_ID: device_config.get(CONF_HOME_ID),
+            },
+        )
+    except PyTado.exceptions.TadoWrongCredentialsException as err:
```

Comment:
Shouldn't we catch this in the config flow and return an error message like the other ways the import can fail?

Reply:
The difference is that it's classified as a device tracker/scanner. As per my current understanding and thorough explanations from Martin, this is all handled in `device_tracker.py`. For this reason, I wrote the import here and is the only current place where I can catch this exception.

Reply:
I mean yes the import belongs here, but this error is a tado error raised somewhere in `async_step_import`. Wouldn't it make more sense to catch the error there and abort with a reason like "invalid_auth"? You would then only to pass a different translation key in `device_tracker.py`

Reply:
There was always a difference between the credentials from the config flow and the device tracker. They could be totally different and never were part of the `async_import_step`. They would never interfere with each other. Since it's thoroughly reviewed by Martin, my understanding is that the device tracker and is separated this way, because it's only invoked if a user has the device tracker YAML configured. It's not part of the default configuration.
Unless this incorrect, a refactor might be justified, but it will consume more time prior to the release.

Reply:
Just catch the exception in the import step around `validate_input` instead and abort.

We need to add a test for the config flow too.